emote/nn/gaussian_policy.py

```diff
@@ -67,7 +67,7 @@ def forward(
         if self.training:
             dist = dists.TransformedDistribution(
                 dists.Independent(dists.Normal(mean, std), 1),
-                RobustTanhTransform(),
+                transforms.TanhTransform(cache_size=1),
```

Comment:
You can do this with the RobustTanhTransform too, it derives from this class.

Reply:
I figured since the values are cached and the transformation is not actually inverted, it doesn't make much of a difference (if I am not misunderstanding how the cache works).

Reply:
I don't know whether to laugh or cry about that, to be honest 🥲

Reply:
I do not think we still need RobustTanhTransform. We do not need to clamp the output of tanh anymore. Even for very large inputs to tanh, the transformation still caches the input and returns the right value for loglikelihood calculations.

Reply:
What happens if we're running forward and backward passes in parallel though? That's the case for clients which doesn't do local inference during training. In that case the cache would not be valid.

Reply:
Wait, how would the cache not be valid? It's always used on the line after it's set, right?

```python
# Transform created
dist = dists.TransformedDistribution(
    dists.Independent(dists.Normal(mean, std), 1),
    transforms.TanhTransform(cache_size=1),
)
# Transform called, cache set
sample = dist.rsample()
# Inverse called in log_prob, cache used
log_prob = dist.log_prob(sample).view(bsz, 1)
```

Reply:
But I do agree with Ali, if we don't use the RobustTanh we should remove it.

Reply:
Ah, it's local. Then yeah nvm me. :D Keep thinking like tensorflow with lots of global state.